orbmain/src/main/java/com/sun/corba/ee/impl/util/Utility.java

```diff
@@ -302,10 +303,14 @@ static Class loadClassOfType(String className, String remoteCodebase,
                     "expected Type.");
             }
 
-            if (expectedTypeClassLoader != null) {
-                loadedClass = expectedTypeClassLoader.loadClass(className);
-            } else {
-                loadedClass = ORBClassLoader.loadClass(className);
+            try {
+                if (expectedTypeClassLoader != null) {
+                    loadedClass = expectedTypeClassLoader.loadClass(className);
+                } else {
+                    loadedClass = ORBClassLoader.loadClass(className);
+                }
+            } catch (Exception e) {
```

Comment:
Could be `ClassNotFoundException` caught only?

Reply:
Yes, I was thinking about that indeed. Wdyt, is it better, or do we risk loosing the type that we have (and might be used) for an optional action to "try" to get an even better type?

Reply:
I worry about non-CNFEx like NPEx being thrown from `ORBClassLoader.loadClass` due to programmer's error to be probably never discovered in logs.
I still would rather catch only specific exception declared by `.loadClass`'es.

Reply:
I just made that change